src/receipt/script_result.rs

```diff
@@ -0,0 +1,34 @@
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
+#[cfg_attr(
+    feature = "serde-types-minimal",
+    derive(serde::Serialize, serde::Deserialize)
+)]
+pub enum ScriptExecutionResult {
+    Success,
+    Revert,
+    Panic,
+    // Generic failure case since any u64 is valid here
+    GenericFailure(u64),
+}
+
+impl From<ScriptExecutionResult> for u64 {
```

Comment:
Maybe use the type aliases created instead of plain `u64`? (`Word`)

Reply:
done